### PR TITLE
[ticket/15440] Add arguments to core.ucp_register_agreement

### PR DESCRIPTION
**Pull request by kasimi**
PHPBB3-15440

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

https://tracker.phpbb.com/browse/PHPBB3-15440

### CODE REVIEW

**Comment by kasimi** [comment]
!set Event

File: phpBB/includes/ucp/ucp_register.php [hunk 1]
```diff
@@ -195,22 +195,32 @@ function main($id, $mode)
...
 			* @event core.ucp_register_agreement
+			* @var	string	tpl_name			Template file
+			* @var	array	template_data		Array with data assigned to the template, read only
+			* @var	array	lang_row			Array with available languages, read only
+			* @var	array	s_hidden_fields		Array with hidden form elements, read only
 			* @since 3.1.6-RC1
+			* @changed 3.2.2-RC1 Added tpl_name, template_data, lang_row, s_hidden_fields
 			*/
-			$phpbb_dispatcher->dispatch('core.ucp_register_agreement');
+			$vars = array('tpl_name', 'template_data', 'lang_row', 's_hidden_fields');
```
Comment (Derky) [inline phpBB/includes/ucp/ucp_register.php]: `template_data` should be changed to `template_vars`. (Don't forget to update it in the comment as well)

**Comment by Derky** [comment]
@kasimi thanks for fixing that. Currently 3 of the 4 variables you add to this event are read-only. I think it would be nicer to do the same thing as we did in https://github.com/phpbb/phpbb/pull/5022#discussion_r154537327

So move the following code after the event to make `$s_hidden_fields` and  `$template_vars` editable by the event:
```
'S_HIDDEN_FIELDS'	=> build_hidden_fields($s_hidden_fields),
```
```
$template->assign_vars($template_vars);
``` 
Then we only have `$lang_row` as a readonly variable.

**Comment by kasimi** [comment]
Moving `$template->assign_vars()` below the event could break extensions that assign template vars. A somewhat radical solution could be to `@deprecate` this event and add a new one above the `$template->assign_vars()` call, but there doesn't seem to be a procedure in place for deprecating events.

**Comment by Derky** [comment]
Good point. I've discussed the options with the other developers. 
Can you mark this event as deprecated and move your changes into a new event? 😃

**Comment by kasimi** [comment]
Done, please have another look.

**Comment by Derky** [comment]
It's squashed and merged

**Pull request closed without merging**